compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -230,6 +230,9 @@ pub fn each_linked_rlib(
                 });
             }
         }
+        if info.dependency_formats.is_empty() {
+            return Err(errors::LinkRlibError::MissingFormat);
```

Comment:
Won't this still result in a compilation error/failed test case for that target as it eventually leads to

https://github.com/rust-lang/rust/blob/77051167083b58c4cd59c658197bcfe6476a7fa5/compiler/rustc_codegen_ssa/src/back/link.rs#L581

Reply:
This branch is only reachable when no crate type is passed to `each_linked_rlib`, which only happens during the LTO code. This code explicitly ignores the error. `link_staticlib` passes in `CrateType::StaticLib` and in addition is only called if `--crate-type staticlib` is used, in which case `dependency_formats` has an entry for `CrateType::StaticLib` and as such is not empty. In addition this code matches what happened before this PR.

Reply:
Thanks!